src/shared/mentionable_text_input.rs

```diff
@@ -676,14 +718,15 @@ impl MentionableTextInput {
             // 3. Virtualization: Only create Widget instances for actually visible items (‎⁠take(MAX_VISIBLE_ITEMS)⁠)
 
             // Performance Improvements:
-            // 1. Reduces processing from over 2,000 items to a maximum of 30 items
-            // 2. Reduces Widget creation from over 2,000 to at most 15 visible Widgets
+            // 1. Reduces processing from over 2,000 items to a reasonable amount
+            // 2. Reduces Widget creation to visible items only
             // 3. Significantly decreases string operations and Widget creation overhead
-            const MAX_VISIBLE_ITEMS: usize = 15;
             let popup = self.cmd_text_input.view(id!(popup));
 
             // Adjust height calculation to include the potential @room item
-            let total_items_in_list = member_count + if "@room".contains(search_text) { 1 } else { 0 };
+            // Use the same condition as when actually adding the @room item
+            let has_room_item = self.can_notify_room && ("@room".contains(search_text) || search_text.is_empty());
+            let total_items_in_list = member_count + if has_room_item { 1 } else { 0 };
```

Comment:
nit: you can convert a bool to 1 or 0 just using `as`:
```suggestion
            let total_items_in_list = member_count + has_room_item as usize;
```